src/components/header.jsx

```diff
@@ -0,0 +1,28 @@
+import React from 'react';
+
+const propTypes = {
+  title: React.PropTypes.string
+};
+
+export default class HeaderComponent extends React.Component {
+
+  //This constructor fires on the instantiation of the component, props is passed in, and we HAVE to super(props) to specify THIS instance (its weird I know), then we can set the state property directly to equal a new object. Equivelant would be getInitialState
+  constructor(props) {
+    super(props);
+    this.state = {
+      header: this.props.title
+    }
+  }
+
+  render() {
+    return (
+      <div className="header">
+        <div className="container">
+          <h1>{ this.props.title }</h1>
```

Comment:
Over here you're accessing the props.title - which luckily right now is exactly the same as this.state.title - so if you wanted to, you could use that instead. In this case, because this component functions as a stateless component, even though it has a state.title property, this isn't a problem - but the recommendation is when you have a stateful component (a component that can internally change it's own state, imagine a button click that can filter for example) - you want your jsx to reference the state properties.